http3/client.go

```diff
@@ -327,6 +327,11 @@ func (c *client) doRequest(
 		res.Uncompressed = true
 	} else {
 		res.Body = respBody
+		if contentLengthStr := res.Header.Get("Content-Length"); contentLengthStr != "" {
```

Comment:
Do we need to set it to -1 if there's no Content-Length header?
See https://github.com/golang/net/blob/4c5254603344ea4a8ae4bed7e296a9588303e14f/http2/transport.go#L2007-L2018.

Reply:
@marten-seemann I think you're right